python/ray/_private/utils.py

```diff
@@ -585,6 +808,28 @@ def resources_from_ray_options(options_dict: Dict[str, Any]) -> Dict[str, Any]:
     return resources
 
 
+def validate_gpu_resource_with_task_and_cluster(
+    task_resources: dict, cluster_resources
+) -> None:
+    """
+    Args:
+        task_resources: construct from remote task/actor's option
+        cluster_resources: the current cluster's resources after ray.init
+    Returns:
+        raise ValueError if task_resources need one type GPU but not specified GPU type,
+        while current cluster_resources contain more than one GPU type
+    """
+    num_gpus = task_resources.get("GPU", None)
+    if num_gpus is not None and num_gpus > 0:
+        task_gpu_types = _get_gpu_types(task_resources)
+        cluster_gpu_types = _get_gpu_types(cluster_resources)
+        if len(task_gpu_types) == 0 and len(cluster_gpu_types) > 1:
+            raise ValueError(
+                "task need GPU but current cluster resources "
+                "contain more than one GPU type."
```

Comment:
can we also add something like "Please specify the acceleator_type..." in the message?